File: test_feature/finance/impl/src/main/java/com/github/klee0kai/stone/test_feature/finance/di/AccountingComponent.java
package com.github.klee0kai.stone.test_feature.finance.di;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.component.Init;

@Component
public interface AccountingComponent {

    AccountingDependencies dependencies();

    AccountingModule accounting();

    @Init
    void initDeps(AccountingDependencies deps);

}

File: test_feature/finance/impl/src/main/java/com/github/klee0kai/stone/test_feature/finance/di/AccountingComponent.kt
package com.github.klee0kai.stone.test_feature.finance.di

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.Init

@Component
interface AccountingComponent {
    fun dependencies(): AccountingDependencies?

    fun accounting(): AccountingModule?

    @Init
    fun initDeps(deps: AccountingDependencies?)
}
